src/overlays/actors/ovl_En_Hg/z_en_hg.c

```diff
@@ -21,48 +52,381 @@ const ActorInit En_Hg_InitVars = {
+                case 5:
+                    this->currentAnimation = 1;
+                    func_800BDC5C(&this->skelAnime, animations, 1);
+                    break;
+                case 6:
+                    gSaveContext.weekEventReg[0x4B] |= 0x20;
+                    Actor_MarkForDeath(&this->actor);
+                    break;
+            }
+        } else {
+            if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount) != 0) {
+                switch (this->currentAnimation) {
+                    case 3:
+                        this->currentAnimation = 4;
+                        func_800BDC5C(&this->skelAnime, animations, 4);
+                        break;
+                    case 5:
+                        this->currentAnimation = 6;
+                        func_800BDC5C(&this->skelAnime, animations, 6);
+                        break;
+                }
+            }
+        }
+        switch (this->currentAnimation) {
+            case 3:
+            case 4:
+                func_800B9010(&this->actor, 0x32B7);
+                break;
+            case 5:
+            case 6:
+                func_800B9010(&this->actor, 0x32B9);
+                break;
+            case 7:
+                if ((this->unk218 == 0) || (this->unk218 == 2)) {
+                    func_800B9010(&this->actor, 0x32B9);
+                }
+                break;
+        }
+        func_800EDF24(&this->actor, globalCtx, actionIndex);
+        return;
+    } else if (globalCtx->csCtx.state == 0) {
+        func_80BCF354(this);
+    }
+    this->cutscenes[3] = 0x63;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF8A0.asm")
+void func_80BCFC0C(EnHg* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Hg_0x80BCF1D0/func_80BCF93C.asm")
+    if (this->actor.colChkInfo.health == 1 && !(fabsf(this->actor.yDistToPlayer) >= 80.0f)) {
+        if (player->stateFlags2 & 0x08000000) {
+            if (D_80BD00C8 == 0) {
+                play_sound(NA_SE_SY_TRE_BOX_APPEAR);
+            }
+            D_80BD00C8 = 1;
+        } else {
+            D_80BD00C8 = 0;
+        }
+        if (globalCtx->msgCtx.unk1202A == 3) {
+            if (globalCtx->msgCtx.unk1202E == 7 && gSaveContext.playerForm == PLAYER_FORM_HUMAN) {
+                if (INV_CONTENT(0x41) == 0x41) {
+                    this->unk218 = 3;
+                } else {
+                    this->unk218 = 1;
+                }
+                func_80BCF88C(this);
+            }
+        } else {
+            if (this->actor.xzDistToPlayer < 60.0f && fabsf(this->actor.yDistToPlayer) < 40.0f) {
+                if ((this->actionFunc != func_80BCF8A0) && (this->actionFunc != func_80BCF95C)) {
+                    if ((gSaveContext.weekEventReg[0x3D] & 2) == 0) {
+                        gSaveContext.weekEventReg[0x3D] |= 2;
+                        this->unk218 = NULL;
```

Comment:
```suggestion
                        this->unk218 = 0;
```
Looks like this is only set to ints, not ptrs.